packages/graph/hash_graph/lib/graph/src/types/mod.rs

```diff
@@ -1,50 +1,74 @@
-//! Descriptions, and implementation logic, of the Types of elements of the graph.
+//! Model types used across datastores
 
-use sqlx::{types::Uuid, FromRow};
+use std::fmt;
 
-#[derive(Debug, FromRow)]
-pub struct DataType {
-    pub version_id: Uuid,
-    pub schema: serde_json::Value,
-    pub created_by: Uuid,
+use uuid::Uuid;
+
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, sqlx::Type, PartialEq, Eq)]
+#[sqlx(transparent)]
+pub struct AccountId(Uuid);
+
+impl AccountId {
+    #[must_use]
+    pub const fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
 }
 
-#[derive(Debug, FromRow)]
-pub struct PropertyType {
-    pub version_id: Uuid,
-    pub schema: serde_json::Value,
-    pub created_by: Uuid,
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, sqlx::Type, PartialEq, Eq)]
+#[sqlx(transparent)]
+pub struct BaseId(Uuid);
+
+impl BaseId {
+    #[must_use]
+    pub const fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
 }
 
-#[derive(Debug, FromRow)]
-pub struct EntityType {
-    pub version_id: Uuid,
-    pub schema: serde_json::Value,
-    pub created_by: Uuid,
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, sqlx::Type, PartialEq, Eq)]
+#[sqlx(transparent)]
+pub struct VersionId(Uuid);
+
+impl VersionId {
+    #[must_use]
+    pub const fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
 }
 
-#[derive(Debug, FromRow)]
-pub struct Entity {
-    pub entity_id: Uuid,
-    pub source_entity_type_version_id: Uuid,
-    pub properties: serde_json::Value,
-    pub created_by: Uuid,
+#[derive(Clone, Debug)]
+pub struct Identifier {
+    pub base_id: BaseId,
+    pub version_id: VersionId,
 }
 
-#[derive(Debug, FromRow)]
-pub struct PropertyTypePropertyTypeReference {
-    pub source_property_type_version_id: Uuid,
-    pub target_property_type_version_id: Uuid,
+impl fmt::Display for Identifier {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // TODO: change this to not just he the debug impl.
+        write!(fmt, "{:?}", self)
+    }
 }
 
-#[derive(Debug, FromRow)]
-pub struct PropertyTypeDataTypeReference {
-    pub source_property_type_version_id: Uuid,
-    pub target_data_type_version_id: Uuid,
+#[repr(transparent)]
+#[derive(Clone, Debug, sqlx::Type, PartialEq, Eq)]
+#[sqlx(transparent)]
+pub struct DataType(serde_json::Value);
+
+impl DataType {
+    #[must_use]
+    pub const fn new(schema: serde_json::Value) -> Self {
+        Self(schema)
+    }
 }
 
-#[derive(Debug, FromRow)]
-pub struct EntityTypePropertyTypeReference {
-    pub source_entity_type_version_id: Uuid,
-    pub target_property_type_version_id: Uuid,
+// TODO: constrain this to only work for valid inner Types.
+#[derive(Clone, Debug)]
+pub struct Qualified<T> {
+    pub id: Identifier,
+    pub inner: T,
+    pub created_by: AccountId,
```

Comment:
I feel like these fields shouldn't be public

Reply:
Addressed in https://github.com/hashintel/hash/pull/734/commits/5b3ca35ed8893e86dbf36467a512d8f8e1af1c9d